_docs/guides/endpoints.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Install Istio for Google Cloud Endpoints Services  
+overview: Guideline to manually integrate Google Cloud Endpoints services with Istio.
+
+order: 42
+layout: docs
+type: markdown
+---
+{% include home.html %}
+
+This document provides a guideline to manually integrate Istio with existing
+Google Cloud Endpoints services. 
+
+## Before you begin
+
+You need to follow the instruction to setup an Endpoints service on GKE [instructions](https://cloud.google.com/endpoints/docs/openapi/get-started-kubernetes-engine)
+After setup, you should be able to get an API key and store it in `ENDPOINTS_KEY` environment variable and the external IP address `EXTERNAL_IP`.
+You may test the service using the following command: 
+```
+curl --request POST       --header "content-type:application/json"       --data '{"message":"hello world"}'       "http://${EXTERNAL_IP}:80/echo?key=${ENDPOINTS_KEY}"
+```
+
+You need to install Istio with [instructions]({{home}}/docs/setup/kubernetes/quick-start.html#google-kubernetes-engine).
+
+## HTTP Endpoints service.
+
+1. Inject the service into Istio mesh using `--includeIPRanges` so that Egress is allowed to call external services directly. 
+Otherwise, ESP won't be able to access Google cloud service control. Follow the [instructions]({{home}}/docs/tasks/traffic-management/egress.html#calling-external-services-directly)
+
+2. After injection, the same testing command by calling ESP directly continue to work. 
+
+3. If you want to access the service through Ingress, setup the following:
```

Comment:
setup the following->create the following ingress service:

may make more sense